Boolder/Models/Circuit.swift

```diff
@@ -233,11 +233,17 @@ extension Circuit {
     }
     
     var firstProblem: Problem? {
-        let query = Table("problems")
+        let queryD = Table("problems")
```

Comment:
this works, but it isn't super DRY
can you refactor it to make it more DRY?
(for instance by extracting a submethod)